### PR TITLE
Block-based swings view vehicle numbers

### PR DESCRIPTION
**Pull request by lemald**
Asana ticket: [Make Swings View more block-oriented for determining vehicle number ](https://app.asana.com/0/1200109164313812/1200341164386661/f)

Once you review #1115 I can merge it and then TypeScript changes will be all that's left on this PR.

### CODE REVIEW

**Comment by phildarnowsky** [comment]
@lemald I'll wait on reviewing this one until #1115 is merged, to help keep things straight.

File: assets/tests/hooks/useVehiclesForBlockIds.test.ts
Comment (phildarnowsky) [inline assets/tests/hooks/useVehiclesForBlockIds.test.ts:8]: I see that we build `VehicleData`s at several points in the test suite as a whole, so let's make a factory for them too. Conceivably we could extract a lot of stuff out of the `Vehicle` factory and use it in both.

**Review by phildarnowsky — CHANGES_REQUESTED** [review]

File: assets/tests/hooks/useVehiclesForBlockIds.test.ts
Comment (lemald) [reply assets/tests/hooks/useVehiclesForBlockIds.test.ts:8]: Ah, yes, I had the inkling feeling I might have forgotten an opportunity to factory-ize things.

Comment (lemald) [reply assets/tests/hooks/useVehiclesForBlockIds.test.ts]: Just pushed up some changes for this.

> Conceivably we could extract a lot of stuff out of the Vehicle factory and use it in both.
So I really _wanted_ to do this, but I couldn't really figure out how the `sequence` values would play along and how we would make sure that we could get a `Vehicle` and `VehicleData` object that we could compare like-for-like. My ultimate solution still leaves some duplication and is perhaps unsatisfying, but it's better than the way it was before.

**Review by phildarnowsky — APPROVED** [review]

**Pull request merged**
